src/backend/marsha/core/api/video.py

```diff
@@ -820,3 +821,105 @@ def initiate_upload(self, request, pk=None):
         Thumbnail.objects.filter(pk=pk).update(upload_state=defaults.PENDING)
 
         return Response(presigned_post)
+
+
+class SharedLiveMediaViewSet(ObjectPkMixin, viewsets.ModelViewSet):
+    """Viewset for the API of the SharedLiveMedia object."""
+
+    permission_classes = [permissions.NotAllowed]
+    queryset = SharedLiveMedia.objects.all()
+    serializer_class = serializers.SharedLiveMediaSerializer
+
+    def get_permissions(self):
+        """Instantiate and return the list of permissions that this view requires."""
+        if self.action in ["create", "list"]:
+            permission_classes = [
+                permissions.IsTokenInstructor
+                | permissions.IsTokenAdmin
+                | permissions.IsParamsVideoAdminThroughOrganization
+                | permissions.IsParamsVideoAdminThroughPlaylist
+            ]
+        elif self.action in ["retrieve"]:
+            permission_classes = [
+                permissions.IsTokenResourceRouteObjectRelatedVideo
+                | permissions.IsRelatedVideoPlaylistAdmin
+                | permissions.IsRelatedVideoOrganizationAdmin
+            ]
+        else:
+            permission_classes = [
+                permissions.IsTokenResourceRouteObjectRelatedVideo
+                & (permissions.IsTokenInstructor | permissions.IsTokenAdmin)
+                | permissions.IsRelatedVideoPlaylistAdmin
+                | permissions.IsRelatedVideoOrganizationAdmin
+            ]
+        return [permission() for permission in permission_classes]
+
+    def list(self, request, *args, **kwargs):
+        """List shared live media through the API."""
+        queryset = self.get_queryset()
+        # If the "user" is just representing a resource and not an actual user profile, restrict
+        # the queryset to tracks linked to said resource
+        user = self.request.user
+        video_id = None
+        if isinstance(user, TokenUser) and (
+            not user.token.get("user")
+            or user.token.get("user", {}).get("id") != user.token.get("resource_id")
+        ):
+            video_id = user.id
+
+        # find the video filter
+        video = request.query_params.get("video")
+        if video is not None:
+            video_id = video
```

Comment:
Doesn't this mean that you can bypass the restriction explained above by setting a video ID? Add a test?

Reply:
Indeed... I have also to check this mechanism on `timed text tracks`